app/src/main/java/org/simple/clinic/summary/SummaryBloodPressureListItem.kt

```diff
@@ -82,7 +85,25 @@ data class SummaryBloodPressureListItem(
     holder.itemView.setOnClickListener { uiEvents.onNext(PatientSummaryBpClicked(measurement)) }
     holder.itemView.isClickable = isEditable
 
-    holder.divider.visibility = if(showDivider) VISIBLE else GONE
+    holder.divider.visibility = if (showDivider) VISIBLE else GONE
+
+    if (displayTime != null) {
+      holder.timeTextView.visibility = VISIBLE
+      holder.timeTextView.text = displayTime
+    } else {
+      holder.timeTextView.visibility = GONE
+    }
+
+    val multipleItemsInThisGroup = displayTime != null
+
+    if (multipleItemsInThisGroup) {
+      if (firstInGroup) holder.itemLayout.setPaddingTop(R.dimen.patientsummary_bp_list_item_first_in_group_top_padding) else holder.itemLayout.setPaddingTop(R.dimen.patientsummary_bp_list_item_multiple_in_group_bp_top_padding)
+      if (showDivider) holder.itemLayout.setPaddingBottom(R.dimen.patientsummary_bp_list_item_last_in_group_bottom_padding) else holder.itemLayout.setPaddingBottom(R.dimen.patientsummary_bp_list_item_multiple_in_group_bp_bottom_padding)
```

Comment:
The reason I did not format these lines is because I felt it is actually easier to understand these nested if/else statements.
Also, Android Studio replaces the xml ids with the actual values, so it won't be that verbose.